app/src/main/java/com/duckduckgo/app/browser/BrowserTabViewModel.kt

```diff
@@ -726,6 +727,37 @@ class BrowserTabViewModel @Inject constructor(
         }
     }
 
+    fun userLongPressedAutocomplete(suggestion: AutoCompleteSuggestion) {
+        when (suggestion) {
+            is AutoCompleteHistorySuggestion -> showRemoveSearchSuggestionDialog(suggestion)
+            is AutoCompleteHistorySearchSuggestion -> showRemoveSearchSuggestionDialog(suggestion)
+            else -> return
+        }
+    }
+
+    private fun showRemoveSearchSuggestionDialog(suggestion: AutoCompleteSuggestion) {
+        command.value = ShowRemoveSearchSuggestionDialog(suggestion)
```

Comment:
Even if not strictly needed, I believe we should post the command on the main thread for extra safety